components/organisms/ContributorProfileTab/contributor-profile-tab.tsx

```diff
@@ -57,78 +69,47 @@ const ContributorProfileTab = ({
 
   const { data: highlights, isError, isLoading, mutate, meta, setPage } = useFetchUserHighlights(login || "");
   const { data: emojis } = useFetchAllEmojis();
-  const [hasHighlights, setHasHighlights] = useState(true);
-
-  const [inputVisible, setInputVisible] = useState(false);
-  const pathnameRef = useRef<string | null>();
 
   const router = useRouter();
 
-  const pathnames = router.pathname.split("/");
-  pathnameRef.current = pathnames[pathnames.length - 1];
-
-  const getCurrentPathName = useMemo(() => {
-    return pathnameRef.current && pathnameRef.current !== "[username]"
-      ? pathnameRef.current
-      : hasHighlights
-      ? "highlights"
-      : "contributions";
-  }, [hasHighlights]);
-
-  const [currentPathname, setCurrentPathname] = useState(getCurrentPathName);
-  const handleTabUrl = (tab: string) => {
-    router.push(`/user/${login}/${tab.toLowerCase()}`);
-    setCurrentPathname(tab.toLowerCase());
-  };
-
-  useEffect(() => {
-    setInputVisible(highlights && highlights.length !== 0 ? true : false);
-    if (login && currentPathname) {
-      setCurrentPathname(getCurrentPathName);
-      router.push(`/user/${login}/${currentPathname}`);
-    }
-    if (login && !hasHighlights && currentPathname) {
-      setCurrentPathname(currentPathname);
-      router.push(`/user/${login}/${currentPathname}`);
-    }
-  }, [login, hasHighlights]);
+  const hasHighlights = highlights ? highlights.length > 0 : false;
```

Comment:
`hasHighlights` needs to stay as a state because the highlight data that updates it is an asynchronous call that might take some time to complete, hence the need to update the `hasHighlights` state from a useEffect.

Reply:
As far as implementation goes, This looks good 🍕. We just have to check on the initial page mount to update the tab query in the URL painted 👍.

Reply:
Thank you for your suggestion. However, as per my understanding, using a state variable for `hasHighlights` may not be necessary in this case because the value of `hasHighlights` is derived from the value of `highlights`, which is returned by the `useSWR` hook. When the value of `highlights` changes, the value of `hasHighlights` will also be updated automatically.

Here's a video of this happening in this feature with logs.

[highlights-hasHighlights-state-issue.webm](https://github.com/open-sauced/app/assets/27003616/0d38e46e-c0b2-43e9-9339-b3b065d14725)

Here's a link for the a 5 min [YouTube chapter](https://youtu.be/GGo3MVBFr1A?t=505&si=P8H_CWjT3G2pKrse). Hope you like it and understand what I am trying to do here.

Reply:
Thanks for sharing this light 🤗🤗🤗... Now I understand👍.

I think my second comment is a case we just have to handle then 👌

>  We just have to check on the initial page mount to update the tab query in the URL painted 👍.

Reply:
I am glad you watched it. It was a good insight. I am taking care of the other suggestion.

Reply:
I have made changes according to the suggestion and marked the PR as Ready for Review.